packages/data-context/test/unit/sources/migration/codegen.spec.ts

```diff
@@ -139,6 +139,21 @@ describe('cypress.config.js generation', () => {
 
     snapshot(generatedConfig)
   })
+
+  it('should maintain both root level and nested non-breaking options during migration', async () => {
+    const projectRoot = getSystemTestProject('migration-e2e-component-default-everything')
+    const config = fs.readJsonSync(path.join(projectRoot, 'cypress.json'))
```

Comment:
Working with files -> Async is better 

```suggestion
    const config = await fs.readJson(path.join(projectRoot, 'cypress.json'))
```